superset/assets/src/SqlLab/utils/reduxStateToLocalStorageHelper.js

```diff
@@ -35,3 +47,18 @@ export default function emptyQueryResults(queries) {
       return updatedQueries;
     }, {});
 }
+
+export function clearQueryEditors(queryEditors) {
+  return queryEditors.map(editor =>
+    // only return selected keys
+    Object.keys(editor).reduce((accumulator, key) => {
```

Comment:
* We could make `PERSISTENT_QUERY_EDITOR_KEYS` a `Set`. 
* `filter` before `reduce` may look a bit  cleaner.

```
Object.keys(editor)
  .filter(key => PERSISTENT_QUERY_EDITOR_KEYS.has(key))
  .reduce((acc, key) => {
     // reuse the same acc object instead of spreading into a new object every iteration.
     // naming a constant from function param will get you pass the eslint rule no param assignment
     const output = acc;
     output[key] = editor[key];
     return output; 
  }, {});
```

Reply:
```
const output = acc;
output[key] = editor[key];
```
this way you will change original copy of accumulator, which makes no sense to assign to a new variable output.

Reply:
Yes, I agree that the purpose of the `eslint` rule is promote immutability over mutation, but since nobody will ever use the intermediate objects I think it might be ok to trade off for less memory usage and garbage collection.